nexus/types/src/deployment.rs

```diff
@@ -97,6 +101,47 @@ impl SledResources {
     }
 }
 
+/// Policy and database inputs to the Reconfigurator planner
+///
+/// The primary inputs to the planner are the parent (either a parent blueprint
+/// or an inventory collection) and this structure. This type holds the
+/// fleet-wide policy as well as any additional information fetched from CRDB
+/// that the planner needs to make decisions.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PlanningInput {
+    /// fleet-wide policy
+    pub policy: Policy,
+
+    /// external IPs allocated to services
+    pub service_external_ips: BTreeMap<TypedUuid<ServiceKind>, ExternalIp>,
+
+    /// vNICs allocated to services
+    pub service_nics: BTreeMap<TypedUuid<ServiceKind>, ServiceNetworkInterface>,
```

Comment:
`ServiceKind` is a little generic -- is there a more specific term for it? I thought these were zone IDs (i.e. `OmicronZoneConfig::id`), is that actually the case?

Reply:
They are zone IDs, yeah. `OmicronZoneKind` maybe?

Reply:
Renamed in [91e3843](https://github.com/oxidecomputer/omicron/pull/5344/commits/91e3843678ea5ebf82c75b148f42ec21d1d23330)